service/history/api/consistency_checker.go

```diff
@@ -101,97 +106,129 @@ func (c *WorkflowConsistencyCheckerImpl) GetCurrentRunID(
 }
 
 func (c *WorkflowConsistencyCheckerImpl) GetWorkflowLease(
+	ctx context.Context,
+	reqClock *clockspb.VectorClock,
+	workflowKey definition.WorkflowKey,
+	lockPriority workflow.LockPriority,
+) (WorkflowLease, error) {
+	return c.getWorkflowLeaseImpl(ctx, reqClock, nil, workflowKey, lockPriority)
+}
+
+// The code below should be used when custom workflow state validation is required.
+// If consistencyPredicate failed (thus detecting a stale workflow state)
+// workflow state will be cleared, and mutable state will be reloaded.
+func (c *WorkflowConsistencyCheckerImpl) GetWorkflowLeaseWithConsistencyCheck(
 	ctx context.Context,
 	reqClock *clockspb.VectorClock,
 	consistencyPredicate MutableStateConsistencyPredicate,
 	workflowKey definition.WorkflowKey,
 	lockPriority workflow.LockPriority,
 ) (WorkflowLease, error) {
-	if reqClock != nil {
-		currentClock := c.shardContext.CurrentVectorClock()
-		if vclock.Comparable(reqClock, currentClock) {
-			return c.getWorkflowLeaseValidatedByClock(
-				ctx,
-				reqClock,
-				currentClock,
-				workflowKey,
-				lockPriority,
-			)
-		}
-		// request vector clock cannot is not comparable with current shard vector clock
-		// use methods below
+
+	return c.getWorkflowLeaseImpl(ctx, reqClock, consistencyPredicate, workflowKey, lockPriority)
+}
+
+func (c *WorkflowConsistencyCheckerImpl) getWorkflowLeaseImpl(
+	ctx context.Context,
+	reqClock *clockspb.VectorClock,
+	consistencyPredicate MutableStateConsistencyPredicate,
+	workflowKey definition.WorkflowKey,
+	lockPriority workflow.LockPriority,
+) (WorkflowLease, error) {
+	if err := c.clockConsistencyCheck(reqClock); err != nil {
+		return nil, err
 	}
 
 	if len(workflowKey.RunID) != 0 {
-		return c.getWorkflowLeaseValidatedByCheck(
-			ctx,
-			consistencyPredicate,
-			workflowKey,
-			lockPriority,
-		)
+		return c.getWorkflowLease(ctx, consistencyPredicate, workflowKey, lockPriority)
 	}
+
 	return c.getCurrentWorkflowContext(
```

Comment:
nit: can you help rename this to `getCurrentWorkflowLease`? I guess it was missed in the previous rename PR (workflowContext -> workflowLease)

Reply:
sure, renamed